feat: test data after redeploy/restart

psturc opened the pull request:
https://issues.jboss.org/browse/AGMS-965
https://issues.jboss.org/browse/AGMS-966

### How to (quickly) verify
1. `git clone --single-branch --branch AGMS-965 https://github.com/aerogear/test-suite && cd test-suite`
1. `npm i && cd backend-admin && npm i`
1. `oc login` as admin to cluster with RHMI + RHMDS installed
1. Start the test`OPENSHIFT_ADMIN_USERNAME=<admin_username> OPENSHIFT_ADMIN_PASSWORD=<admin_password> npm start -- test/data-after-redeployment.js`

psturc commented on common/util/utils.js:
```diff
@@ -23,8 +28,34 @@ const randomString = () =>
-  randomString
+  randomString,
+  getOAuthProxy
 };
```
based on changes from [this PR](https://github.com/aerogear/test-suite/pull/36/files#diff-4a52a5fcf11e0b1648411886550a52c4)

psturc commented on backend/test/data-after-redeployment.js:
```diff
@@ -0,0 +1,249 @@
+    });
+
+    it("should successfully redeploy UPS", async () => {
+      await restartPod({
```
never mind this comment - the function is changed by next commit

b1zzu commented:
@psturc because of the different setup and the need of a different user, wouldn't it be better to put this test in a separate structure (folder/; suite)?
(cc: @jhellar)

psturc commented:
Ah yes, I need to change it. So I see 2 options:
1. Create a new test folder for this single test, like: `test-suite/db-data`
2. Introduce 2 additional env vars (and keep current test folder): `OPENSHIFT_ADMIN_USER`, `OPENSHIFT_ADMIN_PASSWORD`

@b1zzu @jhellar  which one do you prefer?

jhellar commented:
@psturc I would prefer having them in separate test set. I think that @STEPHINRACHEL also moved alert tests to separate test set in her pr. So maybe keep all tests that require admin in one folder eventually?

b1zzu commented:
I also see the second option as a good potential solution especially because there could be more tests in the future which want to perform some actions as an unprivileged user and some as the admin user

psturc commented:
@b1zzu @jhellar after today's discussion I moved the test to `backend-admin` folder and changed env vars from `OPENSHIFT_*` to `OPENSHIFT_ADMIN_*`
So now it makes sense to do the same change for the [alerts test](https://github.com/aerogear/test-suite/pull/36)

jhellar commented on backend-admin/README.md:
```diff
@@ -0,0 +1,30 @@
+
+## Target OpenShift
+
+Tests need to target OpenShift instance with RHMI. For this use:
```
Should we specify here, that the login should be using admin credentials?

psturc replied:
yep, will fix that

jhellar commented on backend-admin/README.md:
````diff
@@ -0,0 +1,30 @@
+
+```
+export OPENSHIFT_ADMIN_USERNAME=...
+export OPENSHIFT_ADMIN_PASSWORD=...
````
These variables are used to login to services, right? This should be possible to do with a regular user. Would it make sense to check services with regular user?

psturc replied:
Yes, regular user could also check the services, but I'm not sure what would be the value in that case...
The next test that we will add to `backend-admin` folder will be [alerts](https://github.com/aerogear/test-suite/pull/36) and that will require `OPENSHIFT_ADMIN_USERNAME`, so I would rather avoid exporting more environment variables than really needed. Wdyt?

jhellar replied:
ok, that makes sense

jhellar commented on common/util/utils.js:
```diff
@@ -24,7 +26,38 @@ const randomString = () =>
+    );
+  }
+
+  const browser = await puppeteer.launch();
```
```suggestion
  const browser = await puppeteer.launch({ slowMo: 50 });
```

jhellar commented on common/util/utils.js:
```diff
@@ -24,7 +26,38 @@ const randomString = () =>
+  }
+  const cookies = await page.cookies();
+  const cookie = cookies.find(c => c.name === "_oauth_proxy").value;
+  await browser.close();
```
If the login fails, browser won't close and the test will never end. This could be a problem when we run the test in CI. Can we wrap the login logic in `try` block with `finally` where we close the browser window?

psturc replied:
ok that makes sense, will add it

jhellar commented on backend-admin/test/data-after-redeployment.js:
```diff
@@ -0,0 +1,256 @@
+        .find(url => url.includes("mobile-security-service"));
+      mssApiEndpoint = `https://${mssHostname}/api`;
+      headers = await getHeaderWithOauthProxyCookie(
+        `${mssApiEndpoint}/apps`,
```
I've tested this against 1.5.1 RHPDS cluster, and the test was not able to log in to MSS. The reason for that was the url. When I removed the `/api/apps` from the end of the url, the login was successful.

psturc replied:
Ok so apparently this endpoint does not work when there are no apps created in MSS. So I will use MSS base URL instead of the `api/apps` endpoint for logging in.
